src/structures/vroom/input/input.h

```diff
@@ -56,6 +56,7 @@ class input {
   std::unique_ptr<vrp> get_problem() const;
   void check_cost_bound() const;
   void set_vehicle_to_job_compatibility();
+  std::vector<int> _matrix_used_index;
```

Comment:
As we just need to remember the set of used indexes, I would go for a `std::unordered_set` here. This would save the need to go through the vector to check for duplicates before insertion: the `if (std::find[...]` checks could be replaced by a simple `insert`.